include/cuco/static_map.cuh

```diff
@@ -197,6 +197,17 @@ class static_map {
              Allocator const& alloc = Allocator{},
              cudaStream_t stream    = 0);
 
+  /**
+   * @brief Construct a fixed-size map with erase capability
+   * empty_key_sentinel and erased_key_sentinel must be different values.
+   */
+  static_map(std::size_t capacity,
+             Key empty_key_sentinel,
+             Value empty_value_sentinel,
+             Key erased_key_sentinel,
```

Comment:
I'm tempted to use simple strong-type wrappers here so it's clearer at the call site what these values mean and there is less chance of confusing them. So something like:
```
namespace sentinel{
   template <typename T>
   struct empty{ T value; };
   
   template <typename T>
   struct erased{T value; };
}

cuco::static_map map{N, 
                     sentinel::empty<Key>{-1}, 
                     sentinel::empty<Value>{-1}, 
                     sentinel::erased<Key>{-2}}; 

```

Reply:
I like this idea and agree that we should do something like this. Could we maybe make it a little simpler though? Maybe something like

```
namespace sentinel {
    template <typename T>
    struct empty_key{T value; };

    template <typename T>
    struct empty_value{T value; };
    
    template <typename T>
    struct erased_key{T value; };
}
```

Reply:
That works for me!